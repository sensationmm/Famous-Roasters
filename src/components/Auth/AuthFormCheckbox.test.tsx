import { MockedProvider } from '@apollo/client/testing'
import { render, waitFor } from '@testing-library/react'
import Form from 'rc-field-form'
import React from 'react'

import { AuthFormCheckbox } from '.'

describe('Auth form checkbox custom auth component', () => {
  const snippet = () => (
    <Form name="testAuthFormItemInput" method="POST">
      <AuthFormCheckbox screenKey="signUp" name="tos" />
    </Form>
  )

  it('Renders correctly', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>{snippet()}</MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
