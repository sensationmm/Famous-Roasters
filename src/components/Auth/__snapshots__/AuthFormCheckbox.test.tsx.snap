// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth form checkbox custom auth component Renders correctly 1`] = `
<div>
  <form
    method="POST"
  >
    <button
      class="flex w-full flex-row items-start cursor-pointer disabled:opacity-20 disabled:cursor-not-allowed"
      data-testid="checkbox"
    >
      <div>
        <svg
          class="undefined fill-brand-black stroke-brand-black"
          fill="inherit"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M19 22H5c-1.7 0-3-1.3-3-3V5c0-1.7 1.3-3 3-3h14c1.7 0 3 1.3 3 3v14c0 1.7-1.3 3-3 3zM5 4c-.6 0-1 .4-1 1v14c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1H5z"
            fill="inherit"
            fill-rule="evenodd"
            stroke="inherit"
            stroke-width="0.25"
          />
        </svg>
      </div>
      <div
        class="flex grow text-left pl-2 text-xs leading-4"
      >
        auth.signUp.label
      </div>
    </button>
  </form>
</div>
`;
